File: evals/behavioral/tasks/rust/fixtures/total-safe-div.bad.rs
// Not total: returns a bare i32, inventing 0 for the b == 0 case. The caller gets
// a usable integer and is never forced to consider the missing answer, so
// `safe_div(1, 0) + 1` type-checks — the invariant is not in the types.
pub fn safe_div(a: i32, b: i32) -> i32 {
    if b == 0 { 0 } else { a / b }
}

File: evals/behavioral/tasks/scala3/fixtures/total-safe-div.bad.scala
// Not total: returns a bare Int, inventing 0 for the b == 0 case. The caller
// gets a usable integer and is never forced to consider the missing answer, so
// `safeDiv(1, 0) + 1` compiles — the invariant is not in the types.
def safeDiv(a: Int, b: Int): Int =
  if b == 0 then 0 else a / b
